cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceStateMachineTest.java

```diff
@@ -873,6 +877,43 @@ private void testStateWithWrongAnnotation(Vertx vertx, VertxTestContext context,
                 .onComplete(result -> defaultStatusHandler(result, context));
     }
 
+    @Test
+    public void testReadyAfterNoRouteToHostException(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+        KafkaRebalance kcRebalance = createKafkaRebalance(KafkaRebalanceState.NotReady, null, null, EMPTY_KAFKA_REBALANCE_SPEC, REASON);
+        this.testStateWithException(vertx, context, 0, CruiseControlEndpoints.REBALANCE, kcRebalance, KafkaRebalanceState.NotReady, KafkaRebalanceAnnotation.none);
+
+        kcRebalance = createKafkaRebalance(KafkaRebalanceState.Ready, null, null, ADD_BROKER_KAFKA_REBALANCE_SPEC, REASON);
+        this.testStateWithException(vertx, context, 0, CruiseControlEndpoints.ADD_BROKER, kcRebalance, KafkaRebalanceState.NotReady, KafkaRebalanceAnnotation.none);
+
+        kcRebalance = createKafkaRebalance(KafkaRebalanceState.Ready, null, null, REMOVE_BROKER_KAFKA_REBALANCE_SPEC, REASON);
+        this.testStateWithException(vertx, context, 0, CruiseControlEndpoints.REMOVE_BROKER, kcRebalance, KafkaRebalanceState.NotReady, KafkaRebalanceAnnotation.none);
+    }
+
+    @Test
+    public void testReadyAfterConnectException(Vertx vertx, VertxTestContext context) throws IOException, URISyntaxException {
+        KafkaRebalance kcRebalance = createKafkaRebalance(KafkaRebalanceState.NotReady, null, null, EMPTY_KAFKA_REBALANCE_SPEC, REASON);
+        this.testStateWithException(vertx, context, 0, CruiseControlEndpoints.REBALANCE, kcRebalance, KafkaRebalanceState.NotReady, KafkaRebalanceAnnotation.none);
+
+        kcRebalance = createKafkaRebalance(KafkaRebalanceState.Ready, null, null, ADD_BROKER_KAFKA_REBALANCE_SPEC, REASON);
+        this.testStateWithException(vertx, context, 0, CruiseControlEndpoints.ADD_BROKER, kcRebalance, KafkaRebalanceState.NotReady, KafkaRebalanceAnnotation.none);
+
+        kcRebalance = createKafkaRebalance(KafkaRebalanceState.Ready, null, null, REMOVE_BROKER_KAFKA_REBALANCE_SPEC, REASON);
+        this.testStateWithException(vertx, context, 0, CruiseControlEndpoints.REMOVE_BROKER, kcRebalance, KafkaRebalanceState.NotReady, KafkaRebalanceAnnotation.none);
+    }
+
+
+    private void testStateWithException(Vertx vertx, VertxTestContext context,
```

Comment:
this methos seems to be exactly the same as `testStateWithWrongAnnotation` but just with a different name. How is it testing that the connection exception is recovered?

Reply:
The `testStateWithException` method is always taking the current state as `ProposalReady` and checking the transition from `NotReady` to `ProposalReady` while for the `testStatewithWrongAnnotations` we have to check for multiple next states since its being used by multiple other tests

Reply:
@ppatierno I have removed the `test` which I was adding named `testStateWithNoRouteToHostEception` since now `CruiseControlRetriableConnectionException` is being thrown in both cases(ConnectExcpetion && testStateWithNoRouteToHostEception) so I guess only one test is enough